### PR TITLE
Read from csv

### PR DESCRIPTION
**Pull request by sarvind7**

### CODE REVIEW

File: src/read_csv.py [hunk 1]
```diff
@@ -1,11 +1,15 @@
 import csv
-countylist = []
-newspaperlist = []
+import sys
+import os
+
+
 def readFile(file):
```
Comment (VeenaArv) [inline src/read_csv.py]: Make this function name more descriptive. Think about what exactly is it reading in.

File: src/read_csv.py [hunk 2]
```diff
@@ -1,11 +1,15 @@
 import csv
-countylist = []
-newspaperlist = []
+import sys
+import os
+
+
 def readFile(file):
+    countylist = []
```
Comment (VeenaArv) [inline src/read_csv.py]: Add a docstring. See https://www.python.org/dev/peps/pep-0257/#:~:text=A%20docstring%20is%20a%20string,special%20attribute%20of%20that%20object.&text=String%20literals%20occurring%20elsewhere%20in%20Python%20code%20may%20also%20act%20as%20documentation.

File: test/test_read_csv.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+from src.read_csv import readFile
+def test_csv_with_one_county():
+    assert readFile('test/test_data/county_to_newspaper_test1.csv') == (["Los Angeles"],["https://www.latimes.com/"])
+
+def test_csv_with_two_counties():
+    assert readFile('test/test_data/county_to_newspaper_test2.csv') == (["Los Angeles", "San Diego"],["https://www.latimes.com/", "https://www.sandiegouniontribune.com/"])
+
+def test_csv_with_three_counties():
```
Comment (VeenaArv) [inline test/test_read_csv.py]: remove this test and corresponding file

File: test/test_read_csv.py [hunk 4]
```diff
@@ -0,0 +1,9 @@
+from src.read_csv import readFile
```
Comment (VeenaArv) [inline test/test_read_csv.py]: can you format your file and make sure you follow the python style guide here: https://www.python.org/dev/peps/pep-0008/

File: src/read_csv.py [hunk 5]
```diff
@@ -1,11 +1,15 @@
...
+    newspaperlist = []
     with open(file, newline='') as csvfile:
         reader = csv.DictReader(csvfile)
         for row in reader:
             countylist.append(row['counties'])
             newspaperlist.append(row['localnewspapers'])
-    totallist = (countylist, newspaperlist)
+    totallist = (countylist,newspaperlist)
     return totallist
```
Comment (VeenaArv) [inline src/read_csv.py]: Here you can just put `return countylist, newspaperlist` and remove the intermediate variable

File: src/read_csv.py [hunk 6]
```diff
@@ -1,11 +1,15 @@
 import csv
-countylist = []
-newspaperlist = []
+import sys
+import os
+
+
 def readFile(file):
+    countylist = []
+    newspaperlist = []
```
Comment (VeenaArv) [inline src/read_csv.py]: Can you use snake case here: `newspaper_list` and `county_list`

File: src/read_csv.py [hunk 7]
```diff
@@ -1,11 +1,15 @@
...
 def readFile(file):
+    countylist = []
+    newspaperlist = []
     with open(file, newline='') as csvfile:
         reader = csv.DictReader(csvfile)
         for row in reader:
             countylist.append(row['counties'])
             newspaperlist.append(row['localnewspapers'])
-    totallist = (countylist, newspaperlist)
+    totallist = (countylist,newspaperlist)
```
Comment (VeenaArv) [inline src/read_csv.py]: space after comma

**Review by VeenaArv — CHANGES_REQUESTED** [review]
Can you also make sure you remove all the .pyc files. I still see one remaining.

File: data/local_newspapers.csv [hunk 8]
```diff
@@ -0,0 +1,52 @@
+county,localnewspapers
```
Comment (VeenaArv) [inline data/local_newspapers.csv]: please add this to a different branch/pr and remove it from this one

File: src/read_csv.py [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+    a list, and the local newspapers in another list and stores
+    the two lists in a tuple."""
+    county_list = []
+    newspaper_list = []
+    with open(file, newline='') as csvfile:
+        reader = csv.DictReader(csvfile)
+        for row in reader:
+            county_list.append(row['counties'])
+            newspaper_list.append(row['localnewspapers'])
+        return (county_list, newspaper_list)
```
Comment (VeenaArv) [inline src/read_csv.py]: you don't need the parentheses. `return county_list, newspaper_list` is inferred as a tuple

File: src/read_csv.py [hunk 10]
```diff
@@ -3,13 +3,16 @@
 import os
 
 
-def readFile(file):
-    countylist = []
-    newspaperlist = []
+def read_in_counties_newspapers(file):
+    """ A function that takes in a file of counties and the
```
Comment (VeenaArv) [inline src/read_csv.py]: Replace docstring with `Takes in a csv file with each row containing a county and newspaper and outputs a list of (county, newspaper) tuples.`

File: src/read_csv.py [hunk 11]
```diff
@@ -3,13 +3,16 @@
 import os
 
 
-def readFile(file):
-    countylist = []
-    newspaperlist = []
+def read_in_counties_newspapers(file):
```
Comment (VeenaArv) [inline src/read_csv.py]: rename to `read_county_to_newspaper_csv`

File: src/read_csv.py [hunk 12]
```diff
@@ -3,13 +3,16 @@
...
+    county_list = []
+    newspaper_list = []
     with open(file, newline='') as csvfile:
         reader = csv.DictReader(csvfile)
         for row in reader:
-            countylist.append(row['counties'])
-            newspaperlist.append(row['localnewspapers'])
-    totallist = (countylist,newspaperlist)
-    return totallist
+            county_list.append(row['counties'])
```
Comment (VeenaArv) [inline src/read_csv.py]: Why are you returning a tuple of list? You should return a list of tuples.

**Review by VeenaArv — CHANGES_REQUESTED** [review]

File: src/read_csv.py (hunk 9 above)
Comment (sarvind7) [reply src/read_csv.py]: done

File: src/read_csv.py (hunk 10 above)
Comment (sarvind7) [reply src/read_csv.py]: done

File: src/read_csv.py (hunk 11 above)
Comment (sarvind7) [reply src/read_csv.py]: done

File: src/read_csv.py (hunk 12 above)
Comment (VeenaArv) [reply src/read_csv.py]: You still need to fix this. You can do this by saying `newspaper_list.append((row['counties], row['localnewspapers']))` and removing county list. Then you will have a list of `[ (county1, newspaper1), (county2, newspaper2)...]`

**Review by VeenaArv — CHANGES_REQUESTED** [review]
One comment. Good Job :)

**Review by VeenaArv — APPROVED** [review]
LGTM

**Pull request merged**